### ISSUE TITLE
provisioning profile picture and buddy pictures

### ISSUE THREAD

**Issue by vieridipaola** [InnovateAsterisk/Browser-Phone#91]
Hi,

I was wondering if it's possible to provision (in a pre js script) the profile picture as well as the buddy pictures (eg. for corporate extensions). The data could be either a base64 string or a url (croppie?).

Thanks

**Comment by InnovateAsterisk** [comment]
Yes, this can be done - profile pictures are saved to the localstorage as Base64.

The basic approach would be to use your pre script, just like you provision the buddies list, except you need to take the following into consideration:

The buddy id needs to be pre-determined. The UID of the buddy that gets written to "XXX-Buddies", would need to be agreed upon out side of this code, so that when you populate the dat, then the uid of the buddy in the buddies list is the same as the base 64 image.

The format of the key in the localstorage is:

`img-<buddy_uid>-extension` or `img-<buddy_uid>-contact` where -"extension" should be used on subscribe buddies, and contacts on regular buddy contacts.

**Also note**, that if your user changes the image, there isn't any way to update your db saved image. This issue and many more issues all go away with XMPP... this is coming :) 
(XMPP will give you a Whatsapp/Skype like experience.)
